core/src/main/java/com/sequenceiq/cloudbreak/service/AbstractOrganizationResourceService.java

```diff
@@ -0,0 +1,174 @@
+package com.sequenceiq.cloudbreak.service;
+
+import static com.sequenceiq.cloudbreak.util.SqlUtil.getProperSqlErrorMessage;
+
+import java.util.Set;
+
+import org.springframework.dao.DataIntegrityViolationException;
+
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUser;
+import com.sequenceiq.cloudbreak.common.type.APIResourceType;
+import com.sequenceiq.cloudbreak.controller.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.controller.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.domain.security.Organization;
+import com.sequenceiq.cloudbreak.domain.security.OrganizationResource;
+import com.sequenceiq.cloudbreak.domain.security.User;
+import com.sequenceiq.cloudbreak.repository.OrganizationResourceRepository;
+import com.sequenceiq.cloudbreak.service.TransactionService.TransactionExecutionException;
+import com.sequenceiq.cloudbreak.service.TransactionService.TransactionRuntimeExecutionException;
+import com.sequenceiq.cloudbreak.service.organization.OrganizationResourceService;
+import com.sequenceiq.cloudbreak.service.organization.OrganizationService;
+import com.sequenceiq.cloudbreak.service.user.UserService;
+
+public abstract class AbstractOrganizationResourceService<T extends OrganizationResource> implements OrganizationResourceService<T> {
+
+    @Override
+    public T create(IdentityUser identityUser, T resource) {
+        return create(identityUser, resource, null);
+    }
+
+    @Override
+    public T create(IdentityUser identityUser, T resource, Long organizationId) {
+        try {
+            return transactionService().required(() -> {
+                try {
+                    prepareCreation(resource);
+                    setOrganization(resource, identityUser, organizationId);
+                    return repository().save(resource);
+                } catch (DataIntegrityViolationException ex) {
+                    String msg = String.format("Error with resource [%s], %s", apiResourceType(), getProperSqlErrorMessage(ex));
+                    throw new BadRequestException(msg);
```

Comment:
`BadRequestException` belongs to REST layer, but the method callable from other layers as well

Reply:
it is callable but in the end of the day it will come from a request anyway and we want our users to know that it's a 403